apis/discoveryengine/v1alpha1/engine_types.go

```diff
@@ -0,0 +1,132 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	refs "github.com/GoogleCloudPlatform/k8s-config-connector/apis/refs/v1beta1"
+	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/apis/k8s/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var DiscoveryEngineEngineGVK = GroupVersion.WithKind("DiscoveryEngineEngine")
+
+// DiscoveryEngineEngineSpec defines the desired state of DiscoveryEngineEngine
+// +kcc:proto=google.cloud.discoveryengine.v1.Engine
+type DiscoveryEngineEngineSpec struct {
+	// Required. The display name of the engine. Should be human readable. UTF-8
+	// encoded string with limit of 1024 characters.
+	DisplayName *string `json:"displayName,omitempty"`
```

Comment:
Just to double check, Is this a required field?

Reply:
Good call - added 	// +required